srcdeps-maven-local-repository/src/main/java/org/l2x6/srcdeps/localrepo/SrcdepsLocalRepositoryManager.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    /**
+     * Delegated to {@link #delegate}
+     *
+     * @see org.eclipse.aether.repository.LocalRepositoryManager#add(org.eclipse.aether.RepositorySystemSession,
+     *      org.eclipse.aether.repository.LocalArtifactRegistration)
+     */
+    @Override
+    public void add(RepositorySystemSession session, LocalArtifactRegistration request) {
+        delegate.add(session, request);
+    }
+
+    /**
+     * Delegated to {@link #delegate}
+     *
+     * @see org.eclipse.aether.repository.LocalRepositoryManager#add(org.eclipse.aether.RepositorySystemSession,
+     *      org.eclipse.aether.repository.LocalMetadataRegistration)
+     */
+    @Override
+    public void add(RepositorySystemSession session, LocalMetadataRegistration request) {
+        delegate.add(session, request);
+    }
+
+    /**
+     * In case the {@link #delegate} does not find the given artifact and the given artifact's version string is a
+     * srcdeps version string, then the version is built from source and returned.
+     *
+     * @see org.eclipse.aether.repository.LocalRepositoryManager#find(org.eclipse.aether.RepositorySystemSession,
+     *      org.eclipse.aether.repository.LocalArtifactRequest)
+     */
+    @Override
+    public LocalArtifactResult find(RepositorySystemSession session, LocalArtifactRequest request) {
+        final LocalArtifactResult result = delegate.find(session, request);
+
+        Artifact artifact = request.getArtifact();
+        String version = artifact.getVersion();
+        if (!result.isAvailable() && SrcVersion.isSrcVersion(version)) {
+            Configuration config = configuration.getConfiguration();
+
+            ScmRepository scmRepo = configuration.findScmRepo(artifact);
+            Path projectBuildDir = scrdepsDir.resolve(scmRepo.getId());
+
+            BuilderIo builderIo = config.getBuilderIo();
+            IoRedirects ioRedirects = IoRedirects.builder() //
+                    .stdin(IoRedirects.parseUri(builderIo.getStdin())) //
+                    .stdout(IoRedirects.parseUri(builderIo.getStdout())) //
+                    .stderr(IoRedirects.parseUri(builderIo.getStderr())) //
+                    .build();
+            BuildRequest buildRequest = BuildRequest.builder() //
+                    .projectRootDirectory(projectBuildDir) //
+                    .scmUrls(scmRepo.getUrls()) //
+                    .srcVersion(SrcVersion.parse(version)) //
+                    .buildArguments(scmRepo.getBuildArguments()) //
+                    .verbosity(config.getVerbosity()) //
+                    .ioRedirects(ioRedirects) //
+                    .build();
+            try {
+                buildService.build(buildRequest);
+
+                /* check once again if the delegate sees the newly built artifact */
+                final LocalArtifactResult newResult = delegate.find(session, request);
+                if (!newResult.isAvailable()) {
+                    log.error(
```

Comment:
How is that possible? By specifying `org.hawkular.accounts` and specifying the `org.hawkular.inventory` repository, for instance?